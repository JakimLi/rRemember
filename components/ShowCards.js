import React, { Component } from 'react';
import {
  StyleSheet,
  ScrollView,
  Image,
  Text
} from 'react-native';
import Card from './Card';

const styles = StyleSheet.create({
  scrollView: {
    paddingTop: 100,
    backgroundColor: '#1a2229'
  }
});

export default ({cards}) => {
  return (
    <ScrollView style={styles.scrollView}
    horizontal={true}
    pageEnbaled={true}>

    {cards.map((card) => {
      return <Card key={card.color + card.point} card={card} />;
    })}
    </ScrollView>
  )
}
