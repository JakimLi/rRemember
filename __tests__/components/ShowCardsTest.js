import 'react-native';
import React from 'react';
import ShowCards from '../../components/ShowCards';

import renderer from 'react-test-renderer';

it('it renders correctly', () => {

  cards = [{
    color: 'Spades',
    point: 'A'
  }];

  const showCardsTree = renderer.create(
    <ShowCards cards={cards} />
  );

  expect(showCardsTree).toMatchSnapshot();
});
