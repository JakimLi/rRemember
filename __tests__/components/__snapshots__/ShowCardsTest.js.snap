exports[`test it renders correctly 1`] = `
<ScrollView>
  <Image
    source={1} />
</ScrollView>
`;
